cpp/src/copying/gather.cu

```diff
@@ -23,16 +24,14 @@ struct dispatch_map_type {
     table_view const& source_table,
     column_view const& gather_map,
     size_type num_destination_rows,
-    bool check_bounds,
-    bool ignore_out_of_bounds,
-    bool allow_negative_indices         = false,
+    out_of_bounds_policy bounds,
+    negative_index_policy neg_indices,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
     cudaStream_t stream                 = 0)
   {
-    std::unique_ptr<table> destination_table;
-
-    if (check_bounds) {
-      cudf::size_type begin = (allow_negative_indices) ? -source_table.num_rows() : 0;
+    if (bounds == out_of_bounds_policy::FAIL) {
+      cudf::size_type begin =
```

Comment:
which is  preferable here `cudf::size_type begin =` or `size_type begin = ` considering it's defined inside `namespace cudf { namespace detail {`

Reply:
I don't think it matters.